.github/workflows/rockset_autoscale.yml

```diff
@@ -0,0 +1,38 @@
+name: Scale Rockset VM
+on:
+  schedule:
+    # Trigger scaling at 8:10am PST and 8:10pm PST on Weekdays (note: chron schedule is in UTC) 
+    # We do 10 mins past the hour to account for any clock skew on the machine this job runs on
+    - cron: "10 3,15 * * 1-5"
+    # Also trigger it on Saturday morning UTC, since that'll be Fri evening PST. This scales it down for the weekend
+    - cron: "10 3 * * 6"
+  # Enable triggering this job manually using the API as well. For testing only
+  push:
+    branches:
+      - "zainr/rockset-autoscale" # to be removed before merging
+  workflow_dispatch:
+
+jobs:
+  autoscale-rockset:
+    runs-on: ubuntu-22.04
+    if: ${{ github.repository == 'pytorch/test-infra' }}
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.8'
+          architecture: x64
+          cache: pip
+
+      - name: Install Python Packages
+        run: |
+          pip3 install rockset==1.0.3
+
+      - name: Scale the Rockset DB up or down based on time of day
+        env:
+          ROCKSET_API_KEY: ${{ secrets.ROCKSET_API_KEY }}
+        run: |
+          python3 .github/scripts/rockset_autoscale.py
```

Comment:
To be honest, I don't really like the idea to add python scripts to the .github directory. What about to use the `/tools/scripts/` directory?

Reply:
I don't feel too strongly about it either ways, so I'm fine with moving it. However, the `.github/scripts` folder has other scripts in it as well.

Perhaps you'd like to write down some guidance around when it's better to place a script in one location vs the other? One of our README.md files would be a good place to put that (or better yet, put it in a new README.md file in `.github/scripts`)

Reply:
I ran into errors trying to import the `tools/scripts/` version of the test from `tools/tests`.  

Perhaps there's a special way to invoke the tests, I couldn't get get the existing `tools/tests/test_fetch_latest_green_commit.py` test to run successfully (it had the same import error). Though I see that that test was moved there a mere 4 weeks ago as well.

If this new test structure requires a special way to be invoked, it would be great to see it documented somewhere. But until then, I'll stick with keeping the script and test together in the tried-and-true pattern used in `.github/scripts`

Error message:
```
(base) zainr@zainr-mbp test-infra % python3 /Users/zainr/test-infra/tools/tests/test_rockset_autoscale.py
Traceback (most recent call last):
  File "/Users/zainr/test-infra/tools/tests/test_rockset_autoscale.py", line 3, in <module>
    from tools.scripts.rockset_autoscale import get_desired_size_at_time, scale_down_size, scale_up_size
ModuleNotFoundError: No module named 'tools.scripts'

```